operator/pkg/values/translation/translation_test.go

```diff
@@ -180,3 +181,382 @@ func TestGetTLSConfigValues(t *testing.T) {
+			values, err := vb.Build()
+			if tt.wantErr {
+				require.NotNil(t, err)
+				return
+			}
+
+			assert.NoError(t, err)
+			assert.Equal(t, tt.want, values)
+		})
+	}
+}
+
+func TestSetScannerV4ComponentValues(t *testing.T) {
+	// using local copies to be able to create references to those constants
+	autoscalingEnabled := platform.ScannerAutoScalingEnabled
+	autoscalingDisabled := platform.ScannerAutoScalingDisabled
+	autoscalingInvalid := platform.AutoScalingPolicy("invalid")
+
+	tests := map[string]struct {
+		component    *platform.ScannerV4Component
+		componentKey string
+		want         chartutil.Values
+		wantErr      bool
+	}{
+		"empty for default component": {
+			component:    &platform.ScannerV4Component{},
+			componentKey: "indexer",
+			want:         chartutil.Values{},
+		},
+		"sets resources": {
+			component: &platform.ScannerV4Component{
+				DeploymentSpec: platform.DeploymentSpec{
+					Resources: &corev1.ResourceRequirements{
+						Requests: corev1.ResourceList{
+							corev1.ResourceCPU:    resource.MustParse("100m"),
+							corev1.ResourceMemory: resource.MustParse("200M"),
+						},
+						Limits: corev1.ResourceList{
+							corev1.ResourceCPU:    resource.MustParse("150m"),
+							corev1.ResourceMemory: resource.MustParse("250M"),
+						},
+					},
+				},
+			},
+			componentKey: "indexer",
+			want: chartutil.Values{
+				"indexer": map[string]interface{}{
+					"resources": map[string]interface{}{
+						"requests": map[string]interface{}{
+							"cpu":    "100m",
+							"memory": "200M",
+						},
+						"limits": map[string]interface{}{
+							"cpu":    "150m",
+							"memory": "250M",
+						},
+					},
+				},
+			},
+		},
+		"uses componentKey as toplevel key": {
+			component: &platform.ScannerV4Component{
+				DeploymentSpec: platform.DeploymentSpec{
+					Resources: &corev1.ResourceRequirements{
+						Requests: corev1.ResourceList{
+							corev1.ResourceCPU: resource.MustParse("100m"),
+						},
+					},
+				},
+			},
+			componentKey: "matcher",
+			want: chartutil.Values{
+				"matcher": map[string]interface{}{
+					"resources": map[string]interface{}{
+						"requests": map[string]interface{}{
+							"cpu": "100m",
+						},
+					},
+				},
+			},
+		},
+		"sets autoscaling if enabled": {
+			component: &platform.ScannerV4Component{
+				Scaling: &platform.ScannerComponentScaling{
+					// using a local copy of platform.ScannerAutoScalingEnabled in order to pass it as a reference
+					// since references to const strings are not allowed in Go
+					AutoScaling: &autoscalingEnabled,
+					MinReplicas: pointers.Int32(1),
+					MaxReplicas: pointers.Int32(3),
+				},
+			},
+			componentKey: "indexer",
+			want: chartutil.Values{
+				"indexer": map[string]interface{}{
+					"autoscaling": map[string]interface{}{
+						"disable":     false,
+						"minReplicas": int32(1),
+						"maxReplicas": int32(3),
+					},
+				},
+			},
+		},
+		"disables autoscaling if disabled": {
```

Comment:
Wouldn't something like
```suggestion
		"autoscaling can be disabled": {
```
be more conscise?

Reply:
yes, done.